### DISCUSSION TITLE
Avoiding the Media coeffs to go to 0

### DISCUSSION THREAD

**Discussion by SaiTarunY4** [facebookexperimental/Robyn#279] [Q&A] [answered]
Hi,
Amazing work being done here with Robyn! I've just started exploring Robyn in detail and have used our data to test out the outcomes. 
I love the model 1 pager which has all the necessary metrics for a MMM, so kudos on that!

I see that sometimes, among all the pareto optimal solutions, the coeffs of some media are being forced to 0. Is there any parameter that can be set to avoid that? I have set the sign to be 'positive'. 

On reading through the glmnet documentation, I see a **'penalty.factor'** parameter that can be used to influence which parameters go to 0. Is this a good place to start? 

Any ideas are appreciated.

**Comment by gufengzhou** [comment, marked as answer]
Hi, thanks for trying out Robyn. Please check out our standard answer to this first https://github.com/facebookexperimental/Robyn/issues/163#issuecomment-937543449

In general, we'll consider include penalty.factor into hyperparameters to be selected automatically. But even with this, if a media has 0 correlation with the dependent, effect will remain 0.
